Exercises.String/Exercises.String/GoogleMapsModel.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Exercises.String
+{
+    class GoogleMapsModel
+    {
+        public Dictionary<string, List<string>> UrlStruct { get; }=new Dictionary<string, List<string>>();
+
+
+        public GoogleMapsModel(string url)
+        {
+            string _url = DecodeUrl(url);
+            int i = 0;
+            while (i < _url.Length)
+            {
+                StringBuilder currentValue = new StringBuilder();
+                StringBuilder currentKey = null;
+                List<string> currentValues = new List<string>();
+
+                if (_url[i] == '&')
+                {
+                    i++;
+                    currentKey = new StringBuilder();
+                    while (_url[i] != '=')
+                    {
+                        currentKey.Append(_url[i]);
+                        i++;
+                    }
+                    i++;
+                }
+
+                while (i < _url.Length && _url[i] != '&')
+                {
+                    if (_url[i] == '|')
+                    {
+                        currentValues.Add(currentValue.ToString());
+                        currentValue.Clear();
+                        i++;
+                    }
+
+                    currentValue.Append(_url[i]);
+                    i++;
+                }
+
+                currentValues.Add(currentValue.ToString());
+
+                if (currentKey == null)
+                {
+                    currentKey = new StringBuilder("url");
+                }
+                UrlStruct.Add(currentKey.ToString(), currentValues);
+            }
+        }
```

Comment:
Too complicated and not optimal 🙂 
1. We strive to readability: using several loops with `i` counter with bunch of checks for each char in method longer than one screen height - too complicated
2. Try to operate not characters but _"components"_ or schema like
`http://    maps.google.com/abc/def?a=a&b=b&c=c&d`
`protocol | host           | path  | query string params = List<{key+value}>
Query string parameter is list of params key+value

Try using `String.Join(...)`, `String.Split(...)`